tensorflow-ops/src/TensorFlow/Gradient.hs

```diff
@@ -108,6 +109,15 @@ type GradientCompatible a =
     -- TODO(fmayle): MaxPoolGrad doesn't support Double for some reason.
     (Num a, OneOf '[ Float, Complex Float, Complex Double ] a)
 
+-- | Types that gradients can be taken w.r.t. (namely 'Tensor' and 'Variable').
+class GradientTarget t where
```

Comment:
To the question of generalizing the name, TensorFlow has `tf.convert_to_tensor` which seems like a similar effect:
https://www.tensorflow.org/api_docs/python/tf/convert_to_tensor
It feels a little odd to put something as specific as `targetZeros` in a general class like that, though.  Up to you; see my comment below in `TensorFlow.Variable`.